src/plugins/midi-piano/midi-piano-display.js

```diff
@@ -0,0 +1,292 @@
+import axios from 'axios';
+import * as Tone from 'tone';
+import { Button } from 'antd';
+import midiPlayerNs from 'midi-player-js';
+import urlUtils from '../../utils/url-utils.js';
+import React, { useEffect, useRef } from 'react';
+import PianoComponent from './piano-component.js';
+import { QuestionCircleOutlined } from '@ant-design/icons';
+import ClientConfig from '../../bootstrap/client-config.js';
+import { useService } from '../../components/container-context.js';
+import { sectionDisplayProps } from '../../ui/default-prop-types.js';
+import PlayIcon from '../../components/icons/media-player/play-icon.js';
+import PauseIcon from '../../components/icons/media-player/pause-icon.js';
+
+export default function MidiPianoDisplay({ content }) {
+
+  const keys = useRef([]);
+  const player = useRef(null);
+  const sampler = useRef(null);
+  const midiData = useRef(null);
```

Comment:
As far as I can see, this also should be a state object (`const [midiData, setMidiData] = useState(null)`)